crates/exex/src/notification.rs

```diff
@@ -0,0 +1,53 @@
+use std::sync::Arc;
+
+use reth_provider::{CanonStateNotification, Chain};
+
+/// Notifications sent to an ExEx.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ExExNotification {
+    /// Chain got committed without a reorg, and only the new chain is returned.
+    ChainCommitted {
+        /// The new chain after commit.
+        new: Arc<Chain>,
+    },
+    /// Chain got reorged, and both the old and the new chains are returned.
+    ChainReorged {
+        /// The old chain before reorg.
+        old: Arc<Chain>,
+        /// The new chain after reorg.
+        new: Arc<Chain>,
+    },
+    /// Chain got reverted, and only the old chain is returned.
+    ChainReverted {
+        /// The old chain before reversion.
+        old: Arc<Chain>,
+    },
+}
+
+impl ExExNotification {
+    /// Returns the committed chain from the [Self::ChainCommitted] variant, if any.
+    pub fn committed_chain(&self) -> Option<Arc<Chain>> {
+        match self {
+            Self::ChainCommitted { new } => Some(new.clone()),
```

Comment:
this missed reorged variant

please use exhaustive match arms

Reply:
right, fixed